Frontend/src/LoginAndRegisterComponents/LoginComponent.jsx

```diff
@@ -0,0 +1,177 @@
+import { useState, useRef } from 'react';
+import LoginForm from './LoginForm';
+import { useTranslation } from 'react-i18next';
+import { GoogleLogin } from '@react-oauth/google';
+import BaseButton from '../Base/BaseButton';
+import axios from 'axios';
+import { jwtDecode } from 'jwt-decode';
+import { useCookies } from 'react-cookie';
+import { modalsActions } from '../Redux/modalsSlice';
+import { useDispatch } from 'react-redux';
+import useAuth from '../Hooks/useAuth';
+import useRefreshToken from '../Hooks/useRefreshToken';
+
+const LoginComponent = () => {
+  const [isValidEmail, setIsValidEmail] = useState(null);
+  const [isValidPassword, setIsValidPassword] = useState(null);
+  const emailRef = useRef(null);
+  const passwordRef = useRef(null);
+  const dispatch = useDispatch();
+  const { setUserCredentials } = useAuth();
+  const [cookies, setCookie] = useCookies(['user']);
+  const { t } = useTranslation();
+  const refresh = useRefreshToken();
+
+  const responseFacebook = (response) => {
+    console.log(response);
+  };
+
+  const handleBlurEmail = () => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    setIsValidEmail(emailRegex.test(emailRef.current.value));
+  };
+
+  const handleBlurPassword = () => {
+    const passwordRegex = /^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*\W)(?!.* ).{8,}$/;
+    setIsValidPassword(passwordRegex.test(passwordRef.current.value));
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (isValidEmail && isValidPassword) {
+      const user = {
+        email: emailRef.current.value.toLowerCase(),
+        password: passwordRef.current.value,
+      };
+      axios
+        .post(
+          'http://localhost:8000/memo_places/token/',
+          {
+            email: `${user.email}`,
+            password: `${user.password}`,
+          },
+          { headers: { 'Content-Type': 'application/json' } },
+        )
+        .then((response) => {
+          const refreshDecoded = jwtDecode(response.data.refresh);
+          setCookie(
+            'user',
+            {
+              user_id: refreshDecoded.user_id,
+              username: refreshDecoded.username,
+              email: refreshDecoded.email,
+              admin: refreshDecoded.admin,
+              master: refreshDecoded.master,
+              refreshToken: refreshDecoded.jti,
+            },
+            {
+              expires: new Date(refreshDecoded.exp * 1000),
+            },
+          );
+          //set as HTTPOnly and secure
```

Comment:
Oraj bo dopóki nie ogarniemy tych ciasteczek to nie ma sensu tego w kodzie trzymać